package.json

```diff
@@ -69,11 +69,13 @@
   ],
   "sideEffects": false,
   "type": "module",
-  "exports": "./index.js",
+  "exports": {
+    "types": "./types/exports.d.ts",
+    "default": "./lib/index.js"
+  },
   "files": [
     "lib/",
-    "index.d.ts",
-    "index.js"
+    "types/"
```

Comment:
What does the generated folder structure look like?
I am not happy about the extra folder. Feels like this is patching a bug that TS should solve.